Added 'allApplications' to XMonad.Prompt.Window

psibi opened the pull request:
### Description

This module allows you to associate the application name of windows with them. I have been using this [module](https://github.com/psibi/dotfiles/blob/30364f1b54a1c093b6eca14e2a263a070627d29a/.xmonad/xmonad.hs#L134) for more than a month and haven't see any problems with it.

### Checklist

  - [x] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - [ ] I tested my changes with [xmonad-testing](https://github.com/xmonad/xmonad-testing)

  - [x] I updated the `CHANGES.md` file

byorgey commented:
This looks fine in general, but I confess I am not quite sure what it actually does or how to use it.  Right now the documentation says "see XMonad.Layout.Tabbed" but I'm guessing that was just copy-pasted from somewhere?  Could you perhaps include some actual documentation / examples of use?

psibi commented:
@byorgey Have pushed a update which hopefully addresses your comments. I have removed the module altogether and have added the actual function in `XMonad.Prompt.Window` with appropriate haddocks.

byorgey commented:
Thanks, this definitely clears it up.  One last thing, can you revert the change to `stack.yaml`?  Then I will merge.  The `stack.yaml` probably should be updated, but that should happen in a separate PR.

psibi commented:
@byorgey The latest push removes the `stack.yaml` file.

byorgey commented:
Thanks!
